tests/Unit/NumericalAlgorithms/Interpolation/Test_RegularGridInterpolant.py

```diff
@@ -0,0 +1,148 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import unittest
+from spectre import Spectral
+from spectre import DataStructures
+from spectre import Interpolation
+from numpy.polynomial.legendre import Legendre
+import numpy as np
+
+
+class TestRegularGrid(unittest.TestCase):
+
+    #random polynomial functions of low order for exact interpolation
+    def polynomial1D(self, points):
+        return points**2 + points + 1
```

Comment:
I think it's easier to make this take and return a 2D array with only 1 row, so it works like the higher-dim functions